File: lib/commands/JustText.dart
import 'package:surake/interfaces/ICommand.dart';
import 'package:surake/main.dart';
import 'package:teledart/src/teledart/model/message.dart';
import 'package:teledart/telegram.dart';

class JustText implements ICommand {
  @override
  Future<void> runner(TeleDartMessage message) {
    final telegram = getIt.get<Telegram>();

    final text = message.text.replaceAll('/$trigger', '');

    if (text.isEmpty) return null;

    if (message.reply_to_message?.message_id != null) {
      telegram.sendMessage(message.chat.id, text,
          reply_to_message_id: message.reply_to_message.message_id);
    } else {
      message.reply(text);
    }

    try {
      telegram.deleteMessage(message.chat.id, message.message_id);
    } catch (err) {
      message.reply('I cant delete the message, please give me ademir');
    }

    return Future(() {});
  }

  @override
  String trigger = 'jt';

  @override
  TriggerMode triggerMode = TriggerMode.COMMAND;
}

File: src/modules/JustText.js
const cmd = "/jt";

const act = async (ctx) => {

    const { message } = ctx;

    const text = message.text.replace(cmd, "");
    const id = message.message_id;
    const reply = message.reply_to_message;
    
    ctx.deleteMessage(id)

    if(!reply) {
        return ctx.reply(text);
    }

    return ctx.reply(text, { reply_to_message_id : reply.message_id })
}

export default { cmd, act };